### DISCUSSION TITLE
NFT page issue

### DISCUSSION THREAD

**Discussion by dbwebapp** [BitCannaGlobal/WeedWallet#73] [General]
On NFT page, the page showing NAN results as following :
Note that I have BitCanna Buddhead on this wallet. It’s on stargaze chain (??) so even on Bitcanna Testnet, it should appears on that page, right ?

![Screenshot 2023-05-24 at 20 46 03](https://github.com/BitCannaGlobal/WeedWallet/assets/96944913/924be102-4271-4b50-8c76-55b7075768fd)

**Comment by dbwebapp** [comment]
After several hard refresh, I got some info but image of my NFT took more than 20 sec to be displayed, this time need to be reduced. What happen if I have 100 NFT to display?
![Screenshot 2023-05-25 at 08 50 55](https://github.com/BitCannaGlobal/WeedWallet/assets/96944913/e4cb76c4-128d-4831-bd5d-ed6199d75e62)

After 24 seconds the image finally appears, look the time below on screenshots: 
![Screenshot 2023-05-25 at 08 56 11](https://github.com/BitCannaGlobal/WeedWallet/assets/96944913/106365db-f4b8-4570-b2fd-267f38d11fe9)

**Comment by atmoner** [comment]
this is related to stargaze api, if their api is slow, the display will be slow....
there is not much to do about it

**Reply by dbwebapp** [reply]
Je comprends, in this case, may be displaying a loading icons "Loading your NFT" so it shows that something is happening, because as a user waiting front of a empty page for more than 20 seconds with no actions can make him go away from this page thinking that there is nothing to display ??

**Reply by TJ-BitCanna** [reply]
Thanks for mentioning. Let's discuss what we can do :)

**Comment by TJ-BitCanna** [comment]
Suggested issue:
- implement loading icon (spinning wheel?) to show that the page is being loaded waiting for a response of the Stargaze API

**Comment by TJ-BitCanna** [comment]
Tested and approved.
